src/resources/assets/assetsApi.ts

```diff
@@ -90,32 +87,121 @@ export class AssetsAPI {
    */
   public delete: AssetDeleteEndpoint;
 
+  private client: CogniteClient;
+
   /** @hidden */
-  constructor(project: string, instance: AxiosInstance, map: MetadataMap) {
+  constructor(
+    client: CogniteClient,
+    project: string,
+    instance: AxiosInstance,
+    map: MetadataMap
+  ) {
+    this.client = client;
     const path = projectUrl(project) + '/assets';
-    this.create = generateCreateEndpoint(instance, path, map, assetChunker);
-    this.list = generateListEndpoint(instance, path, map, true);
-    this.retrieve = generateRetrieveEndpoint(instance, path, map);
-    this.update = generateUpdateEndpoint(instance, path, map);
-    this.search = generateSearchEndpoint(instance, path, map);
+    const transformResponse = this.transformToAssetListObject;
+    this.create = generateCreateEndpoint(
+      instance,
+      path,
+      map,
+      transformResponse,
+      assetChunker
+    );
+    this.list = generateListEndpoint(
+      instance,
+      path,
+      map,
+      true,
+      transformResponse
+    );
+    this.retrieve = generateRetrieveEndpoint(
+      instance,
+      path,
+      map,
+      transformResponse
+    );
+    this.update = generateUpdateEndpoint(
+      instance,
+      path,
+      map,
+      transformResponse
+    );
+    this.search = generateSearchEndpoint(
+      instance,
+      path,
+      map,
+      transformResponse
+    );
     this.delete = generateDeleteEndpoint(instance, path, map);
   }
+
+  public async retrieveSubtree(id: number, depth: number) {
+    const currentDepth: number = 0;
+    const rootAssetList = await this.retrieve([{ id }]);
+    return this.getAssetSubtree(rootAssetList, currentDepth, depth);
+  }
+
+  private transformToAssetListObject = (assets: types.Asset[]) => {
+    const assetArray = assets.map(asset => new Asset(this.client, asset));
+    return new AssetList(this.client, assetArray);
+  };
+
+  private async getAssetSubtree(
+    assets: AssetList,
+    currentDepth: number,
+    depth: number | any
+  ): Promise<AssetList> {
+    const subtree: AssetList = assets;
+    if (depth > currentDepth || depth === Infinity) {
```

Comment:
Don't need to check `|| depth === Infinity`. `Infinity > any number`

Reply:
Fixed